uptrain/core/classes/visuals/dimensionality_reduction.py

```diff
@@ -190,7 +190,7 @@ def base_check(self, inputs, outputs, gts=None, extra_args={}):
                     self.dashboard_name,
                     models=models,
                     features=self.feature_dictn,
-                    file_name=str(count) + "_" + "_".join(list(models.values())),
+                    file_name=self.dashboard_name + "_" + str(count) + "_" + "_".join(list(models.values())),
```

Comment:
@a-r-r-o-w What are the repercussions of changing the file_name here? Does it reflect on the plot name in streamlit? Did you verify nothing broke?

Reply:
@vipgupta I think nothing is broken from whatever I've tested but I could be wrong. The reason why I needed that change is because you can't create 2D and 3D visualizations of either UMAP or TSNE at the same time. You could try using the config from the `test_visuals_dimensionality_reduction.py` file with the older version to test the same.

Let's say you have a config that plots both 2D and 3D UMAP for some data. With the old version, only one of them is visible on the dashboard because the csv file in `uptrain_logs/st_data` is shared between the two and hence there is an overwrite. With this change, you can see both on the dashboard.

Reply:
@a-r-r-o-w if there's a streamlit error when we plot both in 2D and 3D, let's remove one of them from the config for now. Anyways it is unlikely that someone would want both 2D and 3D at the same time.

Reply:
@vipgupta Alright, I'll make the required changes.